src/dependency/Pyegi/Pyegi/installer.py

```diff
@@ -0,0 +1,169 @@
+                lib_links["Packages"].append(lib_link)
+            json.dump(lib_links, open(commons_dir + "lib_links.json", "w"))
+            isdir = os.path.isdir(commons_dir + name_in_commons)
+            if isdir:
+                filename = f"{commons_dir + name_in_commons}/Lib/site-packages/{name_in_commons}.dist-info/RECORD"
+                paths = []
+                with open(filename, "r") as csvfile:
+                    csvreader = csv.reader(csvfile)
+                    for row in csvreader:
+                        paths.append(row[0])
+                for path in paths:
+                    connection_path = "/Lib/site-packages/"
+                    if path[:3] == "../":
+                        connection_path = "/Lib/"
+                        path = path[3:]
+                        if path[:3] == "../":
+                            connection_path = "/"
+                            path = path[3:]
+                    src = commons_dir + name_in_commons + connection_path + path
+                    dst = f"{temp_dir}.venv{connection_path + path}"
+                    create_dirs(dst)
+                    os.symlink(src, dst)
+                    dst = f"{script_path}.venv{connection_path + path}"
+                    create_dirs(dst)
+                    os.symlink(src, dst)
+            else:
+                packages_to_move.append(name_in_commons)
+        phase1_end_time = dt1.now()
+        print("Total phase 1 time: " + str(phase1_end_time - phase1_start_time))
+
+        os.system("poetry install")
+        phase2_start_time = dt1.now()
+        for name_in_commons in packages_to_move:
+            filename = (
+                f"{temp_dir}.venv/Lib/site-packages/{name_in_commons}.dist-info/RECORD"
+            )
+            if exists(filename):
+                paths = []
+                with open(filename, "r") as csvfile:
+                    csvreader = csv.reader(csvfile)
+                    for row in csvreader:
+                        paths.append(row[0])
+                for path in paths:
+                    connection_path = "/Lib/site-packages/"
+                    if path[:3] == "../":
+                        connection_path = "/Lib/"
+                        path = path[3:]
+                        if path[:3] == "../":
+                            connection_path = "/"
+                            path = path[3:]
+                    src = f"{temp_dir}.venv{connection_path + path}"
+                    dst = commons_dir + name_in_commons + connection_path + path
+                    create_dirs(dst)
+                    shutil.move(src, dst)
+                    dst2 = f"{script_path}.venv{connection_path + path}"
+                    create_dirs(dst2)
+                    os.symlink(dst, dst2)
+        phase2_end_time = dt1.now()
+        print("Total phase 2 time: " + str(phase2_end_time - phase2_start_time))
+        os.chdir(os.path.dirname(__file__))
+        shutil.rmtree(temp_dir)
```

Comment:
better to have a check in the first stages of this function, which if there is a temp directory, it should get renewed.
renewing the temp can be a function that we call it in a conditional statement on top and in a non-conditional statement on bottom of `install_pkgs`.

Reply:
ok, nice